### ISSUE TITLE
[patch #2630] Timer macros for ATmega128

### ISSUE THREAD

**Issue by avrs-admin** [avrdudes/avr-libc#691]
Fri 13 Feb 2004 01:52:41 AM CET

I didn't test it yet, but I think this should work for the ATmega128. Is this what <avr/timer.h> was intended for?

example:
// start timer0 in ctc mode, prescaler = 1/8
timer0_ctrl (WGM8_CTC | TC0_CS8);
timer0_set_ocr (250);

[file #6923: timer_h.patch](https://savannah.nongnu.org/patch/download.php?file_id=6923)

This issue was migrated from https://savannah.nongnu.org/patch/?2630

**Comment by avrs-admin** [comment]
Joerg Wunsch <joerg_wunsch>
Fri 13 Feb 2004 10:03:13 PM CET

Actually, <avr/timer.h> has been diminished as being a
bad idea from the beginning.  Sorry for your work, but
why didn't you ask at the mailing list before?

Just yesterday, a request for removal of this file has
been filed... :(

**Comment by avrs-admin** [comment]
Sat 14 Feb 2004 01:58:41 AM CET

> Just yesterday, a request for removal of this file has
> been filed... :(


I know, that's the very reason I wrote the patch.

While I was trying to find a reasonable way to handle the timers I realized that it is almost impossible to have a generic interface that would work for different timers or even for different hardware and still leave you with enough freedom to get everything out of the distinctive timer features.

It was fun doing some inline assembly though, so I simply couldn't stop until I was done. ;)
